pkg/standalone-utils/contracts/relayer/VaultActions.sol

```diff
@@ -95,8 +93,6 @@ abstract contract VaultActions is IBaseRelayerLibrary {
             // parameter.
             _setChainedReferenceValue(outputReferences[i].key, Math.abs(results[outputReferences[i].index]));
         }
-
-        return results;
     }
```

Comment:
We could also check whether the output reference is required before allocating memory for the results instead of always allocating it.

Reply:
Hm, might be a bit overkill. Also, solc likely checks the return data and copies it to memory, even if we don't use it. You can write a simple test to try out this hypothesis.